OpenRA.Mods.Common/Activities/ParadropCargo.cs

```diff
@@ -0,0 +1,69 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class ParadropCargo : Activity
+	{
+		readonly bool unloadAll;
+
+		public ParadropCargo(bool unloadAll)
+		{
+			this.unloadAll = unloadAll;
+		}
+
+		public override Activity Tick(Actor self)
+		{
+			var cargo = self.Trait<Cargo>();
```

Comment:
Please assign these from the constructor (passing `Actor self` as a parameter to query from) instead of looking it up ever tick.